terraform/ec2/common/linux/main.tf

```
@@ -55,6 +55,21 @@ resource "aws_instance" "cwagent" {
   vpc_security_group_ids               = [module.basic_components.security_group]
   associate_public_ip_address          = true
   instance_initiated_shutdown_behavior = "terminate"
+  # Provide a user_data script to disable SSH password authentication
+  user_data = <<-EOT
+    #!/bin/bash
+    # Disable password authentication for SSH
+    sed -i 's/^#*PasswordAuthentication.*/PasswordAuthentication no/' /etc/ssh/sshd_config
+    
+    # Disable challenge-response authentication for SSH
+    sed -i 's/^#*ChallengeResponseAuthentication.*/ChallengeResponseAuthentication no/' /etc/ssh/sshd_config
+    
+    # Disable keyboard-interactive authentication for SSH
+    sed -i 's/^#*KbdInteractiveAuthentication.*/KbdInteractiveAuthentication no/' /etc/ssh/sshd_config
```

Good that you brought this in, we need to make sure to this is set to false too.

You can verify by running `sudo sshd -T` to see what the values are set to.

---

If the issue still persists, a way to overdetermine that password authentication isn't an issue is to just ensure we only allow specific ingress IPs to access the instance, which I believe is being worked on.

Moreover, there could be match/include directives that override the default values in the sshd configuration, so we might want to look out for that as well.

---

We need to run with `sudo`.
```suggestion
    sudo sed -i 's/^#*KbdInteractiveAuthentication.*/KbdInteractiveAuthentication no/' /etc/ssh/sshd_config
```